packages/admin/admin/src/dataGrid/columnsManagement/DataGridColumnsManagement.tsx

```diff
@@ -0,0 +1,181 @@
+// MUI's implementation: https://github.com/mui/mui-x/blob/v7.x/packages/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx
+import {
+    type ComponentsOverrides,
+    type Divider as MuiDivider,
+    type List as MuiList,
+    type ListSubheader,
+    type Theme,
+    useThemeProps,
+} from "@mui/material";
+import { gridColumnVisibilityModelSelector, type GridPinnedColumnPosition, useGridApiContext, useGridSelector } from "@mui/x-data-grid-pro";
+import { type ChangeEvent, useMemo } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { type ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
+import { Body, Divider, List, ListHeader, Root } from "./DataGridColumnsManagement.sc";
+import { DataGridColumnsManagementListItem } from "./DataGridColumnsManagementListItem";
+
+export type DataGridColumnsManagementProps = ThemedComponentBaseProps<{
+    root: "div";
+    body: "div";
+    list: typeof MuiList;
+    listHeader: typeof ListSubheader;
+    divider: typeof MuiDivider;
+}>;
+
+export type DataGridColumnsManagementClassKey = "root" | "body" | "list" | "listHeader" | "divider";
+
+export const DataGridColumnsManagement = (inProps: DataGridColumnsManagementProps) => {
+    const { sx, className, slotProps = {} } = useThemeProps({ props: inProps, name: "CometAdminDataGridColumnsManagement" });
+
+    const apiRef = useGridApiContext();
+    const columnVisibilityModel = useGridSelector(apiRef, gridColumnVisibilityModelSelector);
+    const columns = apiRef.current.getAllColumns();
+
+    const pinnedColumns = apiRef.current.getPinnedColumns();
+
+    const toggleColumn = (event: ChangeEvent<HTMLInputElement>) => {
+        const { name: field } = event.target;
+        const column = columns.find((value) => value.field === field);
+        if (column?.hideable) {
+            const isCurrentlyVisible = columnVisibilityModel[field] !== false;
+
+            const newVisibilityModel = {
+                ...columnVisibilityModel,
+                [field]: !isCurrentlyVisible,
+            };
+
+            apiRef.current.setColumnVisibilityModel(newVisibilityModel);
+        }
+    };
+
+    const pinnedLeftColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return pinnedColumns.left?.includes(column.field) ?? false;
+        });
+    }, [columns, pinnedColumns.left]);
+
+    const scrollableColumns = useMemo(() => {
```

Comment:
`scrollable` seems a bit odd to me... would something like `nonPinnedColumns` or `staticColumns` be better? 🤔

Reply:
sound good. refactored it to `nonPinnedColumns`